packages/react/src/createReactQueryHooks.tsx

```diff
@@ -343,22 +343,23 @@ export function createReactQueryHooks<
       }
       const [path, input] = pathAndInput;
       let isStopped = false;
-      const subscription = client.subscription(
-        path,
-        (input ?? undefined) as any,
-        {
-          error: (err) => {
-            if (!isStopped) {
-              opts.error?.(err);
-            }
-          },
-          next: (res) => {
-            if (res.type === 'data' && !isStopped) {
-              opts.next(res.data);
-            }
-          },
+      const subscription = client.subscription<
+        TRouter['_def']['subscriptions'],
+        TPath,
+        TOutput,
+        inferProcedureInput<TRouter['_def']['subscriptions']>
+      >(path, (input ?? undefined) as any, {
+        error: (err) => {
+          if (!isStopped) {
+            opts.error?.(err);
+          }
         },
-      );
+        next: (res) => {
+          if (res.type === 'data' && !isStopped) {
+            opts.next(res.data);
+          }
+        },
+      });
```

Comment:
Not sure if it's necessary to pass the generics in here for `client.subscription`, but without it I was getting this error:
<details>
 <summary>Error</summary>
 <pre><code>Argument of type 'inferObservableValue<ThenArg<ReturnType<TRouter["_def"]["subscriptions"][TPath]["call"]>>>' is not assignable to parameter of type 'TOutput'.
  'inferObservableValue<ThenArg<ReturnType<TRouter["_def"]["subscriptions"][TPath]["call"]>>>' is assignable to the constraint of type 'TOutput', but 'TOutput' could be instantiated with a different subtype of constraint 'unknown'.
    Type 'unknown' is not assignable to type 'TOutput'.
      'unknown' is assignable to the constraint of type 'TOutput', but 'TOutput' could be instantiated with a different subtype of constraint 'unknown'.
        Type 'inferObservableValue<ThenArg<ReturnType<TRouter["_def"]["subscriptions"][keyof TSubscriptions & string]["call"]>>>' is not assignable to type 'TOutput'.
          'inferObservableValue<ThenArg<ReturnType<TRouter["_def"]["subscriptions"][keyof TSubscriptions & string]["call"]>>>' is assignable to the constraint of type 'TOutput', but 'TOutput' could be instantiated with a different subtype of constraint 'unknown'.
            Type 'unknown' is not assignable to type 'TOutput'.
              'unknown' is assignable to the constraint of type 'TOutput', but 'TOutput' could be instantiated with a different subtype of constraint 'unknown'.
                Type 'inferObservableValue<ThenArg<ReturnType<TRouter["_def"]["subscriptions"][string]["call"]>>>' is not assignable to type 'TOutput'.
                  'inferObservableValue<ThenArg<ReturnType<TRouter["_def"]["subscriptions"][string]["call"]>>>' is assignable to the constraint of type 'TOutput', but 'TOutput' could be instantiated with a different subtype of constraint 'unknown'.
                    Type 'unknown' is not assignable to type 'TOutput'.
                      'unknown' is assignable to the constraint of type 'TOutput', but 'TOutput' could be instantiated with a different subtype of constraint 'unknown'.
                        Type 'unknown' is not assignable to type 'TOutput'.
                          'unknown' is assignable to the constraint of type 'TOutput', but 'TOutput' could be instantiated with a different subtype of constraint 'unknown'.ts(2345)</code></pre>
</details>

Reply:
Let's keep it then! Unnecessary to try to optimize for it. Great job on sorting out the generic!